src/gbt_convertors-recursive-append.pyx

```diff
@@ -0,0 +1,755 @@
+# ===============================================================================
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import json
+from collections import deque
+from os import getpid, remove
+from time import time
+from typing import Any, Deque, Dict, Generator, List, Optional, Tuple
+
+import numpy as np
+import xgboost as xgb
+
+
+class CatBoostNode:
+    def __init__(
+        self,
+        split: Optional[float] = None,
+        value: Optional[List[float]] = None,
+        right: Optional[int] = None,
+        left: Optional[float] = None,
+    ) -> None:
+        self.split = split
+        self.value = value
+        self.right = right
+        self.left = left
+
```

Comment:
It looks like some logic in gbt_convertors-recursive-append duplicates the logic in gbt_convertors. Is it possible to avoid duplication?